src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs

```diff
@@ -186,9 +187,20 @@ public override void ExecuteCmdlet()
                 updateParameters.EnabledState = EnabledState;
             }
 
-            if (this.IsParameterBound(c => c.DisableCertificateNameCheck))
+            // Should not be able to specify both DisableCertificateNameCheck and pass in full BackendPoolsSettings object
+            // But the existence of these fields are optional and not bound to a specific parameter set
+            if (this.IsParameterBound(c => c.BackendPoolsSettings) && this.IsParameterBound(c => c.DisableCertificateNameCheck))
```

Comment:
>if (this.IsParameterBound(c => c.BackendPoolsSettings) && this.IsParameterBound(c => c.DisableCertificateNameCheck)) [](start = 12, length = 116)

Why not use parameterset similar to New-AzFrontDoor?

Reply:
When I tried to do that, I hit issues with PowerShell not being able to resolve the ParameterSet, but I think I wasn't implementing that correctly. Will try again.